File: pkg/pwrgrep/testdata/fixtures/javascript/concurrency.ts
async function f(a, b) {
  // ruleid: js-concurrency
  setTimeout(() => {}, 10);
  // ruleid: js-concurrency
  const w = new Worker("worker.js");
  // ruleid: js-concurrency
  await Promise.all([a, b]);
  // ok: js-concurrency
  await a;
}

File: pkg/pwrgrep/testdata/fixtures/java/Concurrency.java
import java.util.concurrent.Executors;

public class Concurrency {
    void f(java.util.List<Integer> xs) {
        // ruleid: java-concurrency
        new Thread(() -> {}).start();
        // ruleid: java-concurrency
        var pool = Executors.newFixedThreadPool(4);
        // ruleid: java-concurrency
        synchronized (this) {
            xs.clear();
        }
        // ok: java-concurrency
        xs.stream().count();
    }
}
